### PR TITLE
Add validation set option to DFP training

### LINKED ISSUE #544: [FEA]: Add option to provide validation dataset to DFPTraining stage
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

High

### Please provide a clear description of problem this feature solves

The `fit` method in `dfencoder` AutoEncoder class provides option to use a validation dataset:

https://github.com/nv-morpheus/dfencoder/blob/branch-23.01/dfencoder/autoencoder.py#L630

This option, however, is not exposed nor used in the `DFPTraining` stage. 

### Describe your ideal solution

Add option to `DFPTraining` stage for passing validation dataset to underlying `dfencoder` AutoEncoder `fit` method.

### Describe any alternatives you have considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

### PR DESCRIPTION
**Pull request by efajardo-nv**
- Add `validation_size` option to `DFPTraining` stage to create validation dataset from input dataframe. Default is 0.0.
- Add `epochs` option to `DFPTraining` stage. Was previously hard-coded to 30.
- Update `dfp_azure_pipeline.py` and `dfp_duo_pipeline.py` to use validation_size=0.10 in training stage.

Closes #544

### CODE REVIEW

File: examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py [hunk 1]
```diff
@@ -71,13 +75,18 @@ def on_data(self, message: MultiDFPMessage):
...
+        train_df = message.get_meta_dataframe()
 
         # Only train on the feature columns
-        final_df = final_df[final_df.columns.intersection(self._config.ae.feature_columns)]
+        train_df = train_df[train_df.columns.intersection(self._config.ae.feature_columns)]
+        validation_df = None
+
+        # Split into training and validation sets
+        if int(self._validation_size * 100) in range(1, 100):
+            train_df, validation_df = train_test_split(train_df, test_size=self._validation_size, shuffle=False)
```
Comment (mdemoret-nv) [inline examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py:86]: This check should be performed in the constructor to make sure the value is between 0.0 and 1.0. At this point, all we should be doing is checking for `self._validation_size > 0.0`

File: examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py [hunk 2]
```diff
@@ -53,6 +54,9 @@ def __init__(self, c: Config, model_kwargs: dict = None):
         # Update the defaults
         self._model_kwargs.update(model_kwargs if model_kwargs is not None else {})
 
+        self._epochs = epochs
+        self._validation_size = validation_size
```
Comment (mdemoret-nv) [inline examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py:58]: Verify that `validation_size` is between 0.0 and 1.0

File: examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py [hunk 3]
```diff
@@ -31,7 +32,7 @@
 
 class DFPTraining(SinglePortStage):
 
-    def __init__(self, c: Config, model_kwargs: dict = None):
+    def __init__(self, c: Config, model_kwargs: dict = None, epochs=30, validation_size=0.0):
```
Comment (mdemoret-nv) [inline examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py:35]: Need a docstring here for the new options.

Comment (mdemoret-nv) [reply examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py:35]: Make sure to add the range of `validation_size` to the docstring. Ask @dagardner-nv if you havent done this before.

**Review by mdemoret-nv — APPROVED** [review]
Approved with comments.

File: examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py (hunk 3 above)
Comment (efajardo-nv) [reply examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py:35]: Issue #609 created to add docstring to all production DFP code. This docstring will be added as part of a separate PR.
